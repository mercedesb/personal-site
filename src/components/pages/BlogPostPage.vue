<template>
   <div>
     <header :class="'ContentHeader ContentHeader--' + color">
      <div class="ContentHeader-text">
        <h1>{{ page.title }}</h1>
        <h5>{{ formattedPublishDate }}</h5>
      </div>
    </header>
    <main v-if="page.mainContent" class='FlexContainer FlexContainer--justifyCenter'>
      <p class="PageContent">
        <parse-markdown :source="page.mainContent" />
      </p>
    </main>
  </div>
</template>

<script>
const moment = require('moment')

import PageHeader from '../PageHeader.vue'
import CTALink from '../CTALink.vue'

export default {
  components: {
    PageHeader, CTALink
  },
  props: {
    color: {
      type: String,
      default: 'gray'
    }
  },
  computed: {
    page() {
      return this.$store.state.blogPost
    },
    formattedPublishDate() {
      return moment(this.date).format('MMM DD YYYY')
    }
  },
  created () {
    this.$store.dispatch('getBlogPost', this.$route.params.urlSegment)
  }
}
</script>

<style lang="scss">
$header-height: 200px;

.ContentHeader {
  box-shadow: $base-drop-shadow;
  max-height: $header-height;
  display: flex;
  position: relative;
  align-items: center;
  justify-content: center;

  &-text {
    margin: $small-spacing $base-spacing;
    overflow: hidden;
    text-align:center;
  }

  @include background-color;
}
</style>