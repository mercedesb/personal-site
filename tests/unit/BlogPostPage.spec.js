import BlogPostPage from "@/components/pages/BlogPostPage.vue";
import { TestUtility } from "./TestUtility";

describe("BlogPostPage", () => {
  let component;

  const initialProps = {
    color: "blue"
  };

  const shallow = propsData =>
    TestUtility.shallow(BlogPostPage, {
      propsData: {
        ...initialProps,
        ...propsData
      }
    });

  describe("Snapshots", () => {
    describe("with a color passed in", () => {
      it("matches snapshot", () => {
        component = shallow({
          page: TestUtility.blogPosts[0],
          formattedPublishDate: "Feb 17, 2018",
          socialImage: "imageurl"
        });
        expect(component.element).toMatchSnapshot();
      });
    });

    describe("without a color passed in", () => {
      it("matches snapshot", () => {
        component = shallow({
          color: undefined,
          page: TestUtility.blogPosts[0],
          formattedPublishDate: "Feb 17, 2018",
          socialImage: "imageurl"
        });
        expect(component.element).toMatchSnapshot();
      });
    });
  });
});
