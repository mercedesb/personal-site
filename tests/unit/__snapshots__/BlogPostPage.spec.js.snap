// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostPage Snapshots with a color passed in matches snapshot 1`] = `
<main
  class="BlogPost"
>
  <pageheader-stub
    color="blue"
    short="true"
  >
    <template>
      <h1
        role="presentation"
      >
        Feb 17, 2018
      </h1>
    </template>
     
    <template>
      <h2>
        blog post 1 title
      </h2>
    </template>
  </pageheader-stub>
   
  <!---->
</main>
`;

exports[`BlogPostPage Snapshots without a color passed in matches snapshot 1`] = `
<main
  class="BlogPost"
>
  <pageheader-stub
    color="purple"
    short="true"
  >
    <template>
      <h1
        role="presentation"
      >
        Feb 17, 2018
      </h1>
    </template>
     
    <template>
      <h2>
        blog post 1 title
      </h2>
    </template>
  </pageheader-stub>
   
  <!---->
</main>
`;
